src/components/loader/loader.test.tsx

```diff
@@ -30,6 +44,34 @@ test("throws a deprecation warning if the 'aria-label' prop is set", () => {
   loggerSpy.mockRestore();
 });
 
+test.each([
+  ["large", "20px", "8px"],
+  ["medium", "16px", "8px"],
+  ["small", "12px", "6px"],
+  [undefined, "16px", "8px"],
+] as const)(
+  "applies correct styles when size is '%s'",
+  (size, expectedWidth, expectedMarginRight) => {
+    const props = size ? { size } : {};
+
+    render(<Loader {...props} />);
+
+    const squares = screen.getAllByTestId("loader-square");
+    const style = window.getComputedStyle(squares[0]);
+
+    expect(style.width).toBe(expectedWidth);
+    expect(style.marginRight).toBe(expectedMarginRight);
+  },
+);
+
+test("does not render anything when `reduceMotion` is undefined", () => {
+  mockUseMediaQuery.mockReturnValueOnce(undefined);
+
+  render(<Loader />);
+
+  expect(screen.queryByTestId("loader")).not.toBeInTheDocument();
+});
+
 test("when the user disallows animations or their preference cannot be determined, alternative loading text is rendered", () => {
```

Comment:
comment: although this test hasn't been changed, it's always passing

Following the updates I would expect this test to fail when the user does not have reduced motion enabled. It seems that it's finding the visually hidden component with the `getByText("Loading")` instead of the "Loading" text.

We will need to update this test so that it's more specific to what it should be checking. The following should resolve the issues:

```
test("when the user disallows animations, alternative loading text is rendered", () => {
  mockUseMediaQuery.mockReturnValueOnce(false);
  render(<Loader />);
  
  expect(screen.queryByTestId("hidden-label")).not.toBeInTheDocument();
  expect(screen.getByText("Loading")).toBeVisible();
});
```